src/hooks/swap/useSyncTokenDefaults.ts

```diff
@@ -40,45 +42,73 @@ function useDefaultToken(
   return token ?? undefined
 }
 
-export default function useSyncTokenDefaults({
-  defaultInputTokenAddress,
-  defaultInputAmount,
-  defaultOutputTokenAddress,
-  defaultOutputAmount,
-}: TokenDefaults) {
-  const updateSwap = useUpdateAtom(swapAtom)
-  const { chainId } = useWeb3React()
-  const onSupportedNetwork = useOnSupportedNetwork()
+function useDefaultInputToken(
```

Comment:
Why is this necessary? Unless there is a default set, the default input token should be undefined.

Reply:
Great question! So `useDefaultToken` was there from before. My addition was pulling out `useDefaultInputToken`, which used to be some logic defined as a variable which could not be reused. I pulled it out as a hook as I needed it to reuse it in order to get the defaults for the `defaultChainId`